controllers/cache/redis.go

```diff
@@ -43,39 +39,52 @@ func (redis *RedisReconciler) Reconcile() (*lcm.CRStatus, error) {
 		return crStatus, err
 	}
 
-	c, _ := json.Marshal(crStatus)
-	fmt.Println(string(c))
+	crStatus, err = redis.Scale()
+	if err != nil {
+		return crStatus, err
+	}
+
+	crStatus, err = redis.Update(nil)
+	if err != nil {
+		return crStatus, err
+	}
 
 	return crStatus, nil
 }
 
 func (redis *RedisReconciler) Provision() (*lcm.CRStatus, error) {
-	if err := redis.Deploy(); err != nil {
-		return redis.CRStatus, err
+	crStatus, err := redis.Deploy()
+	if err != nil {
+		return crStatus, err
 	}
 
-	if err := redis.Readiness(); err != nil {
-		return redis.CRStatus, err
+	crStatus, err = redis.Readiness()
+	if err != nil {
+		return crStatus, err
 	}
-	return redis.CRStatus, nil
+	return crStatus, nil
 }
 
 func (redis *RedisReconciler) Delete() (*lcm.CRStatus, error) {
 	panic("implement me")
 }
 
 func (redis *RedisReconciler) Scale() (*lcm.CRStatus, error) {
-	panic("implement me")
-}
-
-func (redis *RedisReconciler) ScaleUp(newReplicas uint64) (*lcm.CRStatus, error) {
-	panic("implement me")
-}
-
-func (redis *RedisReconciler) ScaleDown(newReplicas uint64) (*lcm.CRStatus, error) {
-	panic("implement me")
+	crStatus, err := redis.ScaleUpCache()
```

Comment:
I think we should not use such "testing" way to do scale up or down/

We should compare the spec to know what kind of scaling we should do first, and then do the right thing.